api/src/entities/collections.rs

```diff
@@ -25,6 +25,19 @@ pub struct Model {
     pub created_at: DateTimeWithTimeZone,
 }
 
+impl ActiveModelBehavior for ActiveModel {
+    /// Will be triggered before insert / update
+    fn before_save(mut self, _insert: bool) -> Result<Self, DbErr> {
+        if let Some(a) = self.address.as_ref() {
+            if a.starts_with("0x") {
+                self.address = Set(Some(a.to_lowercase()));
+            }
+        }
+
+        Ok(self)
+    }
+}
+
```

Comment:
Certainly! To allow the macro to work with both `Option` and non-`Option` fields, you can modify the attribute macro to take a list of tuples where the first element is the field name and the second element is a boolean indicating whether the field is wrapped in an `Option`. 

Here's the updated code:

```rust
extern crate proc_macro;

use proc_macro::TokenStream;
use quote::quote;
use syn::{parse::Parse, parse_macro_input, ItemImpl, Ident, LitStr, LitBool};

struct SanitizeAddressArgs {
    fields: Vec<(LitStr, LitBool)>,
}

impl Parse for SanitizeAddressArgs {
    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
        let fields = input
            .parse_terminated::<_, syn::token::Comma>(|input| {
                let field_name = input.parse()?;
                let _ = input.parse::<syn::token::Colon>()?;
                let is_option = input.parse()?;
                Ok((field_name, is_option))
            })?
            .into_iter()
            .collect();
        Ok(Self { fields })
    }
}

#[proc_macro_attribute]
pub fn sanitize_address(args: TokenStream, input: TokenStream) -> TokenStream {
    let args = parse_macro_input!(args as SanitizeAddressArgs);
    let input = parse_macro_input!(input as ItemImpl);

    let struct_name = &input.self_ty;
    let field_checks = args.fields.iter().map(|(field_name, is_option)| {
        let field_name_ident = Ident::new(&field_name.value(), field_name.span());
        if is_option.value() {
            quote! {
                if let Some(a) = self.#field_name_ident.as_ref() {
                    if a.starts_with("0x") {
                        self.#field_name_ident = Set(Some(a.to_lowercase()));
                    }
                }
            }
        } else {
            quote! {
                if self.#field_name_ident.starts_with("0x") {
                    self.#field_name_ident = Set(self.#field_name_ident.to_lowercase());
                }
            }
        }
    });

    let expanded = quote! {
        impl #struct_name {
            fn before_save(mut self, _insert: bool) -> Result<Self, DbErr> {
                #(#field_checks)*
                Ok(self)
            }
        }
    };

    TokenStream::from(expanded)
}
```

This code allows you to specify whether each field is wrapped in an `Option` by using a boolean value in the attribute. Here's how you would use the updated macro:

```rust
#[sanitize_address("address": true, "owner": false)]
impl ActiveModel {
    // other methods and fields
}
```

In this example, `"address"` is assumed to be an `Option`, while `"owner"` is not. Modify the code as needed for your specific use case!

Reply:
Example of converting this into a attribute_macro compliments of chatGPT.